internal/kafka/internal/workers/cluster_mgrs/deprovisioning_clusters_mgr.go

```diff
@@ -0,0 +1,128 @@
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/workers"
+	"github.com/google/uuid"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/api"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/metrics"
+	"github.com/golang/glog"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	deprovisioningClustersWorkerType = "deprovisioning_clusters"
+)
+
+// DeprovisioningClustersManager represents a cluster manager that periodically reconciles data plane clusters in deprovisioning state.
+type DeprovisioningClustersManager struct {
+	workers.BaseWorker
+	clusterService         services.ClusterService
+	dataplaneClusterConfig *config.DataplaneClusterConfig
+}
+
+// NewDeprovisioningClustersManager creates a new cluster manager to reconcile data plane clusters in deprovisioning state.
+func NewDeprovisioningClustersManager(reconciler workers.Reconciler, clusterService services.ClusterService, dataplaneClusterConfig *config.DataplaneClusterConfig) *DeprovisioningClustersManager {
+	return &DeprovisioningClustersManager{
+		BaseWorker: workers.BaseWorker{
+			Id:         uuid.New().String(),
+			WorkerType: deprovisioningClustersWorkerType,
+			Reconciler: reconciler,
+		},
+		clusterService:         clusterService,
+		dataplaneClusterConfig: dataplaneClusterConfig,
+	}
+}
+
+// Start initializes the cluster manager to reconcile data plane clusters in deprovisioning state.
+func (m *DeprovisioningClustersManager) Start() {
+	m.StartWorker(m)
+}
+
+// Stop causes the process for reconciling data plane clusters in deprovisioning state to stop.
+func (m *DeprovisioningClustersManager) Stop() {
+	m.StopWorker(m)
+}
+
+func (m *DeprovisioningClustersManager) Reconcile() []error {
+	glog.Infoln("reconciling clusters")
+
+	var errList fleeterrors.ErrorList
+	err := m.processDeprovisioningClusters()
+	if err != nil {
+		errList.AddErrors(err)
+	}
+
+	return errList.ToErrorSlice()
+}
+
+func (m *DeprovisioningClustersManager) processDeprovisioningClusters() error {
+	var errList fleeterrors.ErrorList
+
+	deprovisioningClusters, serviceErr := m.clusterService.ListByStatus(api.ClusterDeprovisioning)
+
+	if serviceErr != nil {
+		errList.AddErrors(serviceErr)
+		return errList
+	}
+
+	glog.Infof("deprovisioning clusters count = %d", len(deprovisioningClusters))
+
+	for i := range deprovisioningClusters {
+		cluster := deprovisioningClusters[i]
+		glog.V(10).Infof("deprovision cluster ClusterID = %s", cluster.ClusterID)
+		metrics.UpdateClusterStatusSinceCreatedMetric(cluster, api.ClusterDeprovisioning)
+		if err := m.reconcileDeprovisioningCluster(&cluster); err != nil {
+			errList.AddErrors(errors.Wrapf(err, "failed to reconcile deprovisioning cluster %s", cluster.ClusterID))
+		}
+	}
+
+	if errList.IsEmpty() {
+		return nil
+	}
+
+	return errList
+}
+
+func (m *DeprovisioningClustersManager) reconcileDeprovisioningCluster(cluster *api.Cluster) error {
+	if m.dataplaneClusterConfig.IsDataPlaneAutoScalingEnabled() {
```

Comment:
I guess part of this logic will be moved out of the deprovisioning reconciler and moved to the future scale down manager right? and even then we will still have a deprovisioning clusters manager

Reply:
The plan was to keep the logic here as well:
- the future scale down manager will perform a similar logic
- and the deprovisioning manager will perform that same logic to be sure that it can safely delete the cluster.

Reply:
It is a sort of double verification that I think we can keep.
- the one in scale down manager reconciler will be used to trigger deprovisioning and marking the cluster as `deprovisioning`
- the one in here will be used to assure that we can safely delete the cluster without causing issues.

Reply:
It might be worth  thinking a little bit about where to perform the sibling logic.
In the case of scale up mgr for example the logic around empty clusters was deleted from any other reconciler and dynamic scale up mgr takes care of that part

Reply:
The plan I've for #1267 is to have a scale down evaluator which performs all the scale down evaluation logic, this will tell us whether a given cluster can be scaled down or no: via the `.ShouldScaleDown()` method of the processor.

Then that logic will be called in the scale down manager by the scale down executor.

And over here, it'll be called in this reconciler as well. 

> It might be worth thinking a little bit about where to perform the sibling logic in next PRs.

So it won't be the sibling logic but it'll be the whole `can we safely scale down` logic being performed here at this part as well. 
The need for that over here is mainly as a safe guard / being extra cautious. 
We could talk more about this on that follow up PR